File: LeetCode/peak-index-in-mountain-array.cpp
int peakIndexInMountainArray(vector<int>& A) {
	int ind = 0;
	while (ind < (A.size() - 1)) {
		if (A[ind] > A[ind + 1]) {
			break;
		}
		ind++;
	}
	return ind;
}

File: LeetCode/peak_index_in_mountain_array.java
public int peakIndexInMountainArray(int[] A) {
  if (A == null || A.length < 3) return -1;
  int low = 0;
  int high = A.length - 1;
  while (low <= high) {
      int mid = low + ((high - low) >> 1);
      if (mid > 0 && mid < (A.length - 1) && 
          A[mid] > A[mid - 1] && A[mid] > A[mid + 1]) {
          return mid;
      } else if (mid > 0 && A[mid - 1] > A[mid]) high = mid - 1;
      else if (mid < (A.length - 1) && A[mid] < A[mid + 1]) low = mid + 1;
      // if (mid <= 0 || mid >= A.length) break;
  }
  return -1;
}
